mathlibtools/lib.py

```diff
@@ -639,6 +639,38 @@ def import_graph(self) -> 'ImportGraph':
         self._import_graph = G
         return G
 
+    def reduce_imports(self, file: str) -> Iterable[List[Tuple[str, str]]]:
+        """
+        An iterator over files with removable imports, for each file yielding
+        a list of removable imports in the format
+        `("removable.import", "source.file")`.
```

Comment:
This is a weird representation. I'd suggest either:
```python
[('src1', ['removable1', 'removable2']),
 ('src2', ['removable3', 'removable4', 'removable5'])]
```
or
```python
[('removable1', 'src1'),
 ('removable2', 'src1'),
 ('removable3', 'src2'),
 ('removable4', 'src2'),
 ('removable5', 'src2')]
```
rather than than the weird structured yet also duplicated representation you've gone for,
```python
[[('removable1', 'src1'),
  ('removable2', 'src1')],
 [('removable3', 'src2'),
  ('removable4', 'src2'),
  ('removable5', 'src2')]]
```

Reply:
Also fine by me, the internal representation is simply the output of networkx rather than a deliberate choice.